common/persistence/dataInterfaces.go

```diff
@@ -1000,6 +1000,18 @@ type (
 		ExclusiveEndTimestamp   time.Time
 	}
 
+	// ListExecutionsRequest is request to ListExecutions
+	ListExecutionsRequest struct {
+		NextPageToken []byte
+		PageSize      int
+	}
+
+	// ListExecutionsResponse is response from ListExecutions
+	ListExecutionsResponse struct {
+		NextPageToken                []byte
+		GetWorkflowExecutionRequests []GetWorkflowExecutionRequest
```

Comment:
this seems a bit weird. Any reason to reuse a request struct in a response struct ?

Reply:
I agree it seems a little weird. But the response I want is exactly this request: domainID and an execution. I could create a type alias to this struct or create a new struct with the same fields?

Reply:
Maybe one reason that its not weird is that after calling this List API the next call will be to get the workflow execution and you will then already have the requests (no need to convert between types).

Reply:
From an API contract perspective, it looks weird. In future, there could be other use cases where the next step is not GetWorkflowExecution(). I am ok with alias or new struct.

Reply:
+1 on using a different name